CHANGELOG.next.toml

```diff
@@ -10,6 +10,14 @@
 # references = ["smithy-rs#920"]
 # meta = { "breaking" = false, "tada" = false, "bug" = false, "target" = "client | server | all"}
 # author = "rcoh"
+[[smithy-rs]]
+message = """
+Fixes the identity resolver types (`credentials_provider()` and `token_provider()`) from `SdkConfig` to have
+a consistent identity cache partition when re-used across different clients.
+"""
+references = ["smithy-rs#3427"]
+meta = { "breaking" = false, "tada" = false, "bug" = true }
+authors = ["aajtodd"]
```

Comment:
I would make this changelog entry `[[aws-sdk-rust]]` since `credentials_provider` and  `token_provider` are AWS-related concepts, rather than general smithy concepts.

In addition, we can also create a changelog entry for `[[smithy-rs]]` mentioned that `SharedIdentityResolver` now respects a cache partition reserved by the passed-in `IdentityResolver`.

Reply:
Sure I can update it as suggested but I want to take issue with this comment for a second:

> since credentials_provider and token_provider are AWS-related concepts, rather than general smithy concepts.


From Smithy and SRA standpoint they are not AWS SDK specific. They are "AWS" concepts in that they originate from AWS but they are not "AWS SDK" only concepts. A customer could use sigv4 in their own model if they wanted (or internally within Amazon/AWS). The way Rust has this organized now is not entirely correct. Sigv4 and token provider support should be in the Smithy runtime (the specific providers like `SsoTokenProvider` or `DefaultCredentialsChain` are of course AWS SDK specific though).

Reply:
> From Smithy and SRA standpoint they are not AWS SDK specific

Good point. To be clear, what I meant was a tooling standpoint that's processing `CHANGELOG.next.toml`. When we write a changelog entry for `[smithy-rs]`, that entry only appears in the release notes in the `smithy-rs` repository, but not in the `aws-sdk-rust` repository. So when those who only use Smithy SDK, but not AWS SDK, see a changelog entry mentioning `credentials_provider` and `token_provider` they don't find the methods because they are defined in the AWS runtime crates & generated by AWS codegen decorators. It can confuse customers.

> The way Rust has this organized now is not entirely correct. Sigv4 and token provider support should be in the Smithy runtime

Interesting. Till we address it, I'd be consistent with the way it has been.